.github/workflows/test.yaml

```diff
@@ -0,0 +1,81 @@
+name: Integration tests
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  build-closure:
```

Comment:
If there's any way  to avoid having these two separate jobs, that would be much preferred.

Since we're trying to dogfood ourselves, we won't be immune to GHAC rate limits, so by storing the binary in the cache, we are at the mercy of those rate limits. If we combine them into one job, we can avoid this since we'll have the executable available locally without needing to fetch it from the cache.

Reply:
The issue I see is that I don't know how to build the closure with the help of the MNC and then use that resulting closure *as* the MNC later on without starting from scratch. We could, of course, build it without caching but that would draw out build times substantially. If you can think of a way around that, I'm all ears.

Reply:
In general, though, I think our setup is generally a bit sloppy here and could probably be streamlined in such a way that we only build things once.